rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderTest.java

```diff
@@ -3191,6 +3194,372 @@ public void saveReceiveHeadersLastTimestamp_not_set() throws IOException {
+        );
+
+        assertThat(provider2.getNextUtxoExpirationCheckpointHeight(), is(Optional.empty()));
+    }
+
+    @Test
+    public void setNextUtxoExpirationCheckpointHeight_after_RSKIP264() {
+        Repository repository = createRepository();
+        Repository track = repository.startTracking();
+
+        BridgeStorageProvider provider1 = new BridgeStorageProvider(
+            track, PrecompiledContracts.BRIDGE_ADDR,
+            config.getNetworkConstants().getBridgeConstants(), activationsAllForks
+        );
+
+        provider1.setNextUtxoExpirationCheckpointHeight(1L);
+        provider1.saveNextUtxoExpirationCheckpointHeight();
+        track.commit();
+
+        track = repository.startTracking();
+
+        BridgeStorageProvider provider2 = new BridgeStorageProvider(
+            track, PrecompiledContracts.BRIDGE_ADDR,
+            config.getNetworkConstants().getBridgeConstants(), activationsAllForks
+        );
+
+        assertThat(provider2.getNextUtxoExpirationCheckpointHeight(), is(Optional.of(1L)));
+    }
+
+    @Test
+    public void setNextUtxoExpirationCheckpointHeight_after_RSKIP264_when_height_is_null() {
+        Repository repository = createRepository();
+        Repository track = repository.startTracking();
+
+        BridgeStorageProvider provider1 = new BridgeStorageProvider(
+            track, PrecompiledContracts.BRIDGE_ADDR,
+            config.getNetworkConstants().getBridgeConstants(), activationsAllForks
+        );
+
+        provider1.setNextUtxoExpirationCheckpointHeight(null);
+        provider1.saveNextUtxoExpirationCheckpointHeight();
+        track.commit();
+
+        track = repository.startTracking();
+
+        BridgeStorageProvider provider2 = new BridgeStorageProvider(
+            track, PrecompiledContracts.BRIDGE_ADDR,
+            config.getNetworkConstants().getBridgeConstants(), activationsAllForks
+        );
+
+        assertThat(provider2.getNextUtxoExpirationCheckpointHeight(), is(Optional.of(0L)));
+    }
+
+    @Test
+    public void setNextUtxoExpirationCheckpointHeight_after_RSKIP264_when_height_is_zero() {
+        Repository repository = createRepository();
+        Repository track = repository.startTracking();
+
+        BridgeStorageProvider provider1 = new BridgeStorageProvider(
+            track, PrecompiledContracts.BRIDGE_ADDR,
+            config.getNetworkConstants().getBridgeConstants(), activationsAllForks
+        );
+
+        provider1.setNextUtxoExpirationCheckpointHeight(0L);
+        provider1.saveNextUtxoExpirationCheckpointHeight();
+        track.commit();
+
+        track = repository.startTracking();
+
+        BridgeStorageProvider provider2 = new BridgeStorageProvider(
+            track, PrecompiledContracts.BRIDGE_ADDR,
+            config.getNetworkConstants().getBridgeConstants(), activationsAllForks
+        );
+
+        assertThat(provider2.getNextUtxoExpirationCheckpointHeight(), is(Optional.of(0L)));
```

Comment:
When testing the `set` for an invalid value I think we could also add a check that the repository was not called

`verify(repository, never()).addStorageBytes(
            eq(PrecompiledContracts.BRIDGE_ADDR),
            eq(NEXT_UTXO_EXPIRATION_CHECKPOINT_HEIGHT),
            any()
        );`

In this case for example it's not clear if the value was set or not, since the value you passed was 0 and that's the value you are getting

Reply:
repository Is Not a Mock, Hence, verify() Can't Be Used

Reply:
But you could mock it, right? I see it mocked for example in `saveNextUtxoExpirationCheckpointHeight_before_RSKIP264` test

Reply:
No It Can't Be Mocked For setter Tests